bson/src/test/unit/org/bson/types/ObjectIdTest.java

```diff
@@ -140,6 +176,55 @@ public void testDateCons() {
         assertEquals(new Date().getTime() / 1000, new ObjectId(new Date()).getDate().getTime() / 1000);
     }
 
+    @Test
+    public void testRandomConstructors() {
+        assertNotEquals(new ObjectId(new Date(1_000)), new ObjectId(new Date(1_000)));
+        assertEquals(new ObjectId(new Date(1_000), 1), new ObjectId(new Date(1_000), 1));
+        assertEquals(new ObjectId(1_000, 1), new ObjectId(1_000, 1));
+
+        assertEquals("00000001", new ObjectId(new Date(1_000)).toHexString().substring(0, 8));
+        assertEquals("00000001", new ObjectId(new Date(1_000), 1).toHexString().substring(0, 8));
+        assertEquals("7fffffff", new ObjectId(Integer.MAX_VALUE, 1).toHexString().substring(0, 8));
+
+        assertThrows(NullPointerException.class, () -> new ObjectId(null, Integer.MAX_VALUE));
+        assertThrows(IllegalArgumentException.class, () -> new ObjectId(new Date(1_000), Integer.MAX_VALUE));
+        assertThrows(IllegalArgumentException.class, () -> new ObjectId(Integer.MAX_VALUE, Integer.MAX_VALUE));
+        assertThrows(IllegalArgumentException.class, () -> new ObjectId((ByteBuffer) null));
+        assertThrows(IllegalArgumentException.class, () -> new ObjectId(ByteBuffer.allocate(11)));
+    }
+
+    /**
+     * MethodSource for valid ByteBuffers containing an ObjectID at the current position.
+     */
+    public static List<ByteBuffer> inputBuffers() {
+        byte[] data = new byte[12];
+        for (byte i = 0; i < data.length; ++i) {
+            data[i] = i;
+        }
+
+        List<ByteBuffer> result = new ArrayList<>();
+        result.add(ByteBuffer.wrap(data));
+        result.add(ByteBuffer.wrap(data).order(ByteOrder.LITTLE_ENDIAN));
+        result.add(setPosition(ByteBuffer.allocateDirect(data.length).put(data), 0));
+        result.add(setPosition(ByteBuffer.allocateDirect(data.length).put(data).order(ByteOrder.LITTLE_ENDIAN), 0));
+        result.add(setPosition(ByteBuffer.allocate(2 * data.length).put(data), 0));
+        result.add(setPosition(ByteBuffer.allocate(2 * data.length).put(new byte[12]).put(data), 12));
+        return result;
+    }
+
+    @ParameterizedTest
+    @MethodSource(value = "inputBuffers")
+    public void testByteBufferCons(final ByteBuffer input) {
```

Comment:
Let’s use a more explicit test name to make the purpose clearer at a glance.
```suggestion
    public void testByteBufferConstructor(final ByteBuffer input) {
```